### PR TITLE
The NPC PR 2/3 OK

### PR DESCRIPTION
**Pull request by Prakxo**

### CODE REVIEW

File: include/m_common_data.h [hunk 1]
```diff
@@ -234,7 +235,7 @@ typedef struct CommonData {
     /* 0x104B0 */ UNK_TYPE1 unk_104B0[0xE8];
     /* 0x10598 */ mQst_not_saved_c quest;
     /* 0x105A0 */ u32 scene_from_title_demo;
-    /* 0x105A4 */ UNK_TYPE1 unk_105A4[0xF0];
+    /* 0x105A4 */ NpsSchedule npcSchedule[ANIMAL_NUM_MAX];
```
Comment (AngheloAlf) [inline include/m_common_data.h:238]:
```suggestion
    /* 0x105A4 */ NpsSchedule npcSchedule[ANIMAL_NUM_MAX];
```

File: include/m_npc_schedule.h [hunk 2]
```diff
@@ -0,0 +1,44 @@
+#ifndef M_NPC_SCHEDULE_H
+#define M_NPC_SCHEDULE_H
+
+#include "ultra64.h"
+#include "m_npc.h"
+
+typedef struct NpsScheduleData {
+    /* 0x00 */ u32 type;
```
Comment (AngheloAlf) [inline include/m_npc_schedule.h:8]: Is this type a `NpsScheduleType`?

Comment (Prakxo) [reply include/m_npc_schedule.h:8]: Yeah it is, I didn't realise

Comment (AngheloAlf) [reply include/m_npc_schedule.h:8]: Would using `NpsScheduleType` instead of `u32` match?

Comment (Prakxo) [reply include/m_npc_schedule.h:8]: It does match, so I'll push a change

**Pull request merged**